src/Machine.js

```diff
@@ -238,17 +247,28 @@ class Machine {
   _includeSource(source, context, buffer, once, evaluated) {
 
     // path is an expression, evaluate it
-    const includePath = evaluated ? source : this.expression.evaluate(
+    let includePath = evaluated ? source : this.expression.evaluate(
         source, this._mergeContexts(this._globalContext, context)
-      );
+      ).trim();
 
     // if once flag is set, then check if source has alredy been included
     if (once && this._includedSources.has(includePath)) {
       this.logger.debug(`Skipping source "${includePath}": has already been included`);
       return;
     }
 
-    const reader = this._getReader(includePath);
+    let reader = this._getReader(includePath);
+    let needCache = false;
+    if (this.fileCache.toBeCached(includePath) && this.fileCache.isCachedReader(reader)) {
+        let result;
+        if ((result = this.fileCache.findFile(includePath)) && !this.fileCache.isCacheFileOutdate(result)) {
+          // change reader to local reader
+          includePath = result;
+          reader = this.readers.file;
+        } else {
+          needCache = true;
+        }
+    }
     const includePathParsed = reader.parsePath(includePath);
```

Comment:
It provide its local path, If I understood you correctly. 
` let result;
        if ((result = this.fileCache.findFile(includePath)) && !this.fileCache.isCacheFileOutdate(result)) {
          // change reader to local reader
          includePath = result;
          reader = this.readers.file;
        } else {
          needCache = true;
        }`
before reader is changed, local path is set up